### PR TITLE
Keycloak User Provider Migration

### PR DESCRIPTION
**Pull request by jacob6838**
# PR Details

## Description

This PR includes instructions for migrating from a disconnected Keycloak and Postgres solution the new Keycloak custom user provider connected solution. This also includes a fix for querying for keycloak users in keycloak 21.1

The new migration steps are described in resources/keycloak/README.md

## How Has This Been Tested?

These migration steps were tested on the dev cdot environment, and were successful, and have have been slightly refined afterwards.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [x] My changes require updates to the documentation:
  - [x] I have updated the documentation accordingly.
- [ ] My changes require updates and/or additions to the unit tests:
  - [ ] I have modified/added tests to cover my changes.
- [x] All existing tests pass.

### CODE REVIEW

File: resources/keycloak/README.md [hunk 1]
```diff
@@ -10,3 +10,67 @@ The `realm.json` file included in this project initializes Keycloak with a sampl
 ## Keycloak Theme
 
 A sample keycloak theme is provided in the `sample_theme.jar` file. This is a sample theme generated using [Keycloakify](https://github.com/CDOT-CV/keycloakify-starter), to use a custom theme put a generated .jar file in this directory and then update the `KEYCLOAK_LOGIN_THEME_NAME` with the name of the new .jar file.
+
+## Migration Steps
+
+This section describes the steps required to add this custom user provider to an existing cvmanager deployment. If followed correctly, there will be no action required by users (other than possible having local users re-set their credentials, more on that later), and no user data will be lost.
```
Comment (drewjj) [inline resources/keycloak/README.md]: "other than possible having local users" is this meant to be "possibly"

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ The JPO CV Manager was originally developed for the Google Cloud Platform and a
 
 - Keycloak is used for the CV Manager webapp's authentication.
 - The Keycloak pod requires a `realm.json` file in the folder: `./resources/keycloak/` to startup with the proper configurations. It also requires a login theme that can be modified and generated using the [keycloakify](https://github.com/keycloakify/keycloakify) forked repository in resources/keycloak/keycloakify. The theme will be automatically generated when using the docker image provided but can also be built using instructions found in the keycloakify folder.
+- To migrate an environment from a disconnected keycloak/postgres deployment the Keycloak postgres user provider, follow the steps described in [resources/keycloak/README.md](./resources/keycloak/README.md#migration-steps)
```
Comment (drewjj) [inline README.md]: I keep rereading this sentence and it doesn't seem right to me lol. Grammaticaly.

**Review by drewjj — APPROVED** [review]
Just a few grammatical details but everything looks great! I like the documentation.

File: resources/keycloak/README.md (hunk 1 above)
Comment (jacob6838) [reply resources/keycloak/README.md]: fixed

File: README.md (hunk 2 above)
Comment (jacob6838) [reply README.md]: fixed

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**